### PR TITLE
Add title field to vulnerability entity

### PR DESCRIPTION
**Pull request by udgover**

### CODE REVIEW

**Review by tomchop — CHANGES_REQUESTED** [review]
LGTM. One small not however: can we update the docstring to detail the fields and what we expect from them? Maybe even adding a validator to the name field (CVE-blah) would be a good idea and help disambiguate how these entities should be used

File: core/schemas/entity.py [hunk 1]
```diff
@@ -179,13 +185,25 @@ class CourseOfAction(Entity):
...
         r"(?P<pre>\W?)(?P<search>CVE-\d{4}-\d{4,7})(?P<post>\W?)"
-    )
+    ))
 }
 
+def validate_entity(ent: Entity) -> bool:
+    if ent.type in TYPE_VALIDATOR_MAP:
+        return TYPE_VALIDATOR_MAP[ent.type](ent) is True
+    elif ent.type in dict(REGEXES_ENTITIES):
+        for field, regex in REGEXES_ENTITIES[ent.type]:
```
Comment (tomchop) [inline core/schemas/entity.py:200]: Don't you get an unpacking error here? `REGEXES_ENTITIES[ent.type]` is a tuple, and you're iterating twice (`for` + `field, regex`)

File: core/schemas/entity.py [hunk 2]
```diff
@@ -179,13 +185,25 @@ class CourseOfAction(Entity):
...
+            if regex.match(getattr(ent, field)):
+                return True
+        return False
+    else:
+        return False
```
Comment (tomchop) [inline core/schemas/entity.py:205]:
```suggestion
    return False
```

**Review by tomchop — CHANGES_REQUESTED** [review]

File: core/schemas/entity.py (hunk 1 above)
Comment (udgover) [reply core/schemas/entity.py]: You're right!

Comment (udgover) [reply core/schemas/entity.py]: Fixed in d1721b3cbae7adf9097fc56df7f6e0a12ee410a7

File: core/schemas/entity.py (hunk 2 above)
Comment (udgover) [reply core/schemas/entity.py]: Fixed in d1721b3cbae7adf9097fc56df7f6e0a12ee410a7

**Review by tomchop — APPROVED** [review]

**Pull request merged**